src/ec/curve25519/ops.rs

```diff
@@ -84,16 +81,32 @@ impl Point {
     }
 
     pub fn to_bytes(&self) -> [u8; ELEM_LEN] {
-        let mut bytes = [0u8; ELEM_LEN];
-        unsafe { GFp_x25519_ge_tobytes(&mut bytes, self); }
+        point_to_bytes(&self.x, &self.y, &self.z)
+    }
+}
 
-        bytes
+fn point_to_bytes(x: &Elem, y: &Elem, z: &Elem) -> [u8; ELEM_LEN] {
+    let mut recip = [0; ELEM_LIMBS];
+    let mut x_over_z = [0; ELEM_LIMBS];
+    let mut y_over_z = [0; ELEM_LIMBS];
+    let mut bytes = [0; ELEM_LEN];
+
+    unsafe {
+        GFp_fe_invert(&mut recip, z);
+        GFp_fe_mul(&mut x_over_z, x, &recip);
+        GFp_fe_mul(&mut y_over_z, y, &recip);
+        GFp_fe_tobytes(&mut bytes, &y_over_z);
     }
+    bytes[31] ^= unsafe { GFp_fe_isnegative(&x_over_z) << 7 };
```

Comment:
I folded this suggestion into the "Port [...]" commit [here](https://github.com/briansmith/ring/pull/522/commits/4c27993ade28a99ab498fc8fcdedf9dea0803cb8#diff-d26b986b38fde7051f0848091dad3a9bR105).